src/frontend/src/binder/expr/mod.rs

```diff
@@ -935,17 +935,31 @@ pub fn bind_data_type(data_type: &AstDataType) -> Result<DataType> {
                 .collect::<Result<Vec<_>>>()?,
             types.iter().map(|f| f.name.real_value()).collect_vec(),
         ),
-        AstDataType::Custom(qualified_type_name) if qualified_type_name.0.len() == 1 => {
-            // In PostgreSQL, these are not keywords but pre-defined names that could be extended by
-            // `CREATE TYPE`.
-            match qualified_type_name.0[0].real_value().as_str() {
+        AstDataType::Custom(qualified_type_name) => {
+            let idents = qualified_type_name
+                .0
+                .iter()
+                .map(|n| n.real_value())
+                .collect_vec();
+            let name = if idents.len() == 1 {
+                idents[0].as_str() // `int2`
+            } else if idents.len() == 2 && idents[0] == PG_CATALOG_SCHEMA_NAME {
+                idents[1].as_str() // `pg_catalog.text`
+            } else {
+                return Err(new_err().into());
+            };
+
+            // In PostgreSQL, these are non-keywords or non-reserved keywords but pre-defined
+            // names that could be extended by `CREATE TYPE`.
+            match name {
```

Comment:
Should we support all other types in `RW_TYPE_DATA` (or `for_all_base_types`)?

https://github.com/risingwavelabs/risingwave/blob/c791c8d22a2d1889c09c26f59c03503497703288/src/common/src/types/postgres_type.rs#L17-L28